core/templates/pages/topic-editor-page/questions-tab/topic-questions-tab.component.spec.ts

```diff
@@ -0,0 +1,237 @@
+    expect($scope.skillIdToRubricsObject).toEqual(skillIdToRubricsObject);
+    expect($scope.allSkillSummaries).toEqual(allSkillSummaries);
+    expect($scope.getSkillsCategorizedByTopics).toBe(categorizedSkillsDictData);
+    expect($scope.getUntriagedSkillSummaries)
+      .toBe(untriagedSkillSummariesData);
+    expect($scope.canEditQuestion).toBe(false);
+    expect($scope.misconceptions).toEqual([]);
+    expect($scope.questionIsBeingUpdated).toBe(false);
+    expect($scope.questionIsBeingSaved).toBe(false);
+    expect($scope.emptyMisconceptionsList).toEqual([]);
+  });
+
+  it('should setFocus on selectSkillField when screen loads', function() {
+    $window.onload();
+
+    expect(focusManagerService.setFocus).toHaveBeenCalledWith(
+      'selectSkillField');
+  });
+
+  it('should initialize tab when topic is initialized', function() {
+    const topicRights = TopicRights.createInterstitialRights();
+    const allSkillSummaries = subtopic1.getSkillSummaries();
+    $scope.allSkillSummaries = null;
+    $scope.topicRights = null;
+    $scope.topic = null;
+
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBeNull();
+
+    topicInitializedEventEmitter.emit();
+
+    expect($scope.allSkillSummaries).toEqual(allSkillSummaries);
+    expect($scope.topicRights).toEqual(topicRights);
+    expect($scope.topic).toBe(topic);
+  });
+
+  it('should initialize tab when topic is reinitialized', function() {
+    const topicRights = TopicRights.createInterstitialRights();
+    const allSkillSummaries = subtopic1.getSkillSummaries();
+    $scope.allSkillSummaries = null;
+    $scope.topicRights = null;
+    $scope.topic = null;
+
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBeNull();
+
+
+    topicInitializedEventEmitter.emit();
+    expect($scope.allSkillSummaries).toEqual(allSkillSummaries);
+    expect($scope.topicRights).toEqual(topicRights);
+    expect($scope.topic).toBe(topic);
+    $scope.allSkillSummaries = null;
+    $scope.topicRights = null;
+    $scope.topic = null;
+    topicReinitializedEventEmitter.emit();
+
+    expect($scope.allSkillSummaries).toEqual(allSkillSummaries);
+    expect($scope.topicRights).toEqual(topicRights);
+    expect($scope.topic).toBe(topic);
+  });
+
+  it('should unsubscribe when component is destroyed', function() {
+    spyOn(ctrl.directiveSubscriptions, 'unsubscribe');
+
+    ctrl.$onDestroy();
+
+    expect(ctrl.directiveSubscriptions.unsubscribe).toHaveBeenCalled();
+  });
+
+  it('should reinitialize questions list when a skill is selected', function() {
+    spyOn(qls, 'resetPageNumber');
+    spyOn(qls, 'getQuestionSummariesAsync');
+
+    $scope.reinitializeQuestionsList('1');
+
+    expect($scope.selectedSkillId).toEqual('1');
+    expect(qls.resetPageNumber).toHaveBeenCalled();
+    expect(qls.getQuestionSummariesAsync).toHaveBeenCalledWith('1', true, true);
```

Comment:
Try to check the final state instead, rather than that a function was called.

Remember that you need a pre-check too in order to ensure that a change actually happened.

Reply:
DONE